### PR TITLE
Fix Replication Job

### PR DESCRIPTION
**Pull request by apc999**
Continuation of https://github.com/Alluxio/alluxio/pull/13362

### CODE REVIEW

File: job/server/src/main/java/alluxio/job/util/JobUtils.java [hunk 1]
```diff
@@ -147,7 +147,7 @@ public static void loadBlock(URIStatus status, FileSystemContext context, long b
     }
     // TODO(bin): remove the following case when we consolidate tier and medium
     // since there is only one element in the set, we take the first element in the set
-    String medium = pinnedLocation.iterator().next();
+    String medium = pinnedLocation.isEmpty() ? "" : pinnedLocation.iterator().next();
```
Comment (bf8086) [inline job/server/src/main/java/alluxio/job/util/JobUtils.java:150]: What is the behavior when medium is empty?

File: job/server/src/test/java/alluxio/job/plan/replicate/ReplicateDefinitionTest.java [hunk 2]
```diff
@@ -269,32 +269,23 @@ public void runTaskNoBlockWorker() throws Exception {
...
-    mTestStatus.getFileInfo().setMediumTypes(Sets.newHashSet("MEM"));
-    byte[] input = BufferUtils.getIncreasingByteArray(0, (int) TEST_BLOCK_SIZE);
-    TestBlockInStream mockInStream =
-        new TestBlockInStream(input, TEST_BLOCK_ID, input.length, false,
-            BlockInStreamSource.NODE_LOCAL);
-    TestBlockOutStream mockOutStream =
-        new TestBlockOutStream(ByteBuffer.allocate(MAX_BYTES), TEST_BLOCK_SIZE);
-    BlockWorkerInfo localBlockWorker = new BlockWorkerInfo(LOCAL_ADDRESS, TEST_BLOCK_SIZE, 0);
-    runTaskReplicateTestHelper(Lists.newArrayList(localBlockWorker), mockInStream, mockOutStream);
-    assertArrayEquals(input, mockOutStream.getWrittenData());
```
Comment (bf8086) [inline job/server/src/test/java/alluxio/job/plan/replicate/ReplicateDefinitionTest.java:283]: Is array compare no longer needed for the test?

File: job/server/src/main/java/alluxio/job/util/JobUtils.java (hunk 1 above)
Comment (apc999) [reply job/server/src/main/java/alluxio/job/util/JobUtils.java:150]: medium is actually empty typically. it is non-empty only when the file is pinned to a medium which is rare

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 3]
```diff
@@ -173,7 +173,7 @@ private RpcUtils() {} // prevent instantiation
       }
     } catch (Exception e) {
       logger.warn("Exit(stream) (Error): {}: {}, Error={}", methodName,
-          String.format(description, args), e);
```
Comment (apc999) [inline core/server/common/src/main/java/alluxio/RpcUtils.java:176]: I saw from logs
```
2021-05-07 20:06:31,676 WARN  ShortCircuitBlockReadHandler - Exit(stream) (Error): OpenBlock: Session=0, Request=null, Error={}
alluxio.exception.BlockDoesNotExistException: blockId 5553258511 not found
	at alluxio.worker.grpc.ShortCircuitBlockReadHandler$1.call(ShortCircuitBlockReadHandler.java:90)
	at alluxio.worker.grpc.ShortCircuitBlockReadHandler$1.call(ShortCircuitBlockReadHandler.java:69)
	at alluxio.RpcUtils.streamingRPCAndLog(RpcUtils.java:163)
	at alluxio.worker.grpc.ShortCircuitBlockReadHandler.onNext(ShortCircuitBlockReadHandler.java:69)
	at alluxio.worker.grpc.ShortCircuitBlockReadHandler.onNext(ShortCircuitBlockReadHandler.java:36)
	at io.grpc.stub.ServerCalls$StreamingServerCallHandler$StreamingServerCallListener.onMessage(ServerCalls.java:255)
	at io.grpc.ForwardingServerCallListener.onMessage(ForwardingServerCallListener.java:33)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.messagesAvailableInternal(ServerCallImpl.java:309)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.messagesAvailable(ServerCallImpl.java:292)
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1MessagesAvailable.runInContext(ServerImpl.java:765)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

File: job/server/src/test/java/alluxio/job/plan/replicate/ReplicateDefinitionTest.java (hunk 2 above)
Comment (apc999) [reply job/server/src/test/java/alluxio/job/plan/replicate/ReplicateDefinitionTest.java:283]: I added the assert on output array back

File: job/server/src/test/java/alluxio/job/plan/replicate/ReplicateDefinitionTest.java [hunk 4]
```diff
@@ -269,32 +269,23 @@ public void runTaskNoBlockWorker() throws Exception {
...
+        byte[] input = BufferUtils.getIncreasingByteArray(0, (int) TEST_BLOCK_SIZE);
+        TestBlockInStream mockInStream =
+            new TestBlockInStream(input, TEST_BLOCK_ID, input.length, false,
+                BlockInStreamSource.NODE_LOCAL);
+        TestBlockOutStream mockOutStream =
+            new TestBlockOutStream(ByteBuffer.allocate(MAX_BYTES), TEST_BLOCK_SIZE);
+        BlockWorkerInfo localBlockWorker = new BlockWorkerInfo(LOCAL_ADDRESS, TEST_BLOCK_SIZE, 0);
+        runTaskReplicateTestHelper(Lists.newArrayList(localBlockWorker), mockInStream,
+            mockOutStream);
+        assertEquals(TEST_BLOCK_SIZE, mockInStream.getBytesRead());
```
[the remark this answers is not in the document]
Comment (apc999) [reply job/server/src/test/java/alluxio/job/plan/replicate/ReplicateDefinitionTest.java:287]: added back

[the remark this answers is not in the document]
Comment (bf8086) [reply job/server/src/test/java/alluxio/job/plan/replicate/ReplicateDefinitionTest.java:287]: Add message here as well?

**Review by bf8086 — APPROVED** [review]
LGTM.

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**